src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiImporter.cs

```diff
@@ -0,0 +1,370 @@
+			var deprecated = element.XGetAttribute ("deprecated");
+			var is_static = element.XGetAttribute ("static") == "true";
+			var jni_signature = element.XGetAttribute ("jni-signature");
+
+			var model = new JavaInterfaceModel (package, nested_name, visibility, deprecated, is_static, jni_signature);
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				model.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				model.PropertyBag.Add ("deprecated-since", dep);
+
+			if (element.Element ("typeParameters") is XElement tp)
+				ParseTypeParameters (model.TypeParameters, tp);
+
+			foreach (var child in element.Elements ()) {
+				switch (child.Name.LocalName) {
+					case "field":
+						model.Fields.Add (ParseField (model, child));
+						break;
+					case "implements":
+						model.Implements.Add (ParseImplements (child));
+						break;
+					case "method":
+						if (child.XGetAttribute ("synthetic") != "true")
+							model.Methods.Add (ParseMethod (model, child));
+						break;
+				}
+			}
+
+			return model;
+		}
+
+		public static JavaMethodModel ParseMethod (JavaTypeModel type, XElement element)
+		{
+			var method = new JavaMethodModel (
+				javaName: element.XGetAttribute ("name"),
+				javaVisibility: element.XGetAttribute ("visibility"),
+				javaAbstract: element.XGetAttributeAsBool ("abstract"),
+				javaFinal: element.XGetAttributeAsBool ("final"),
+				javaStatic: element.XGetAttributeAsBool ("static"),
+				javaReturn: element.XGetAttribute ("return"),
+				javaParentType: type,
+				deprecated: element.XGetAttribute ("deprecated"),
+				jniSignature: element.XGetAttribute ("jni-signature"),
+				isSynthetic: element.XGetAttributeAsBool ("synthetic"),
+				isBridge: element.XGetAttributeAsBool ("bridge"),
+				returnJni: element.XGetAttribute ("jni-return"),
+				isNative: element.XGetAttributeAsBool ("native"),
+				isSynchronized: element.XGetAttributeAsBool ("synchronized"),
+				returnNotNull: element.XGetAttributeAsBool ("return-not-null")
+			);
+
+			if (element.Element ("typeParameters") is XElement tp)
+				ParseTypeParameters (method.TypeParameters, tp);
+
+			foreach (var child in element.Elements ("parameter"))
+				method.Parameters.Add (ParseParameter (method, child));
+			foreach (var child in element.Elements ("exception"))
+				method.Exceptions.Add (ParseException (child));
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				method.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				method.PropertyBag.Add ("deprecated-since", dep);
+
+			return method;
+		}
+
+		public static JavaConstructorModel ParseConstructor (JavaTypeModel type, XElement element)
+		{
+			var method = new JavaConstructorModel (
+				javaName: element.XGetAttribute ("name"),
```

Comment:
Looking at an `bin/Build*/api/api-30.xml.in`, there is no `//constructor/@abstract` attribute.

Additionally, there are no instances of `//constructor[@final = 'true']`, and in every instance I've looked at, `//constructor/@name` is the same as the containing`//class/@name` value.  I'm thus not sure there's a need for `javaName`, `javaFinal`, or `javaAbstract`.

For that matter, `api-30.xml.in` has no `//constructor[@static='true']`.  In theory there *could* be one, in the form of static initializers; I wonder if `class-parse` skips them, or if `android.jar` just doesn't have any.